### PR TITLE
[Validator] add class name to the cache key

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |  #20685 
| License       | MIT
| Doc PR        | 

Adding the class name to the cache key to avoid collision

### CODE REVIEW

**Review by dunglas — CHANGES_REQUESTED** [review]
Can you also add a test with a relation between ChildA and ChilB?

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
👍

File: src/Symfony/Component/Validator/Tests/Fixtures/Entity.php [hunk 1]
```diff
@@ -33,13 +33,20 @@ class Entity extends EntityParent implements EntityInterfaceB
...
+     * @Assert\Valid
+     */
+    public $childB;
     protected $lastName;
     public $reference;
     public $reference2;
     private $internal;
     public $data = 'Overridden data';
     public $initialized = false;
-
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Fixtures/Entity.php]: this line should be kept

Comment (Simperfit) [inline src/Symfony/Component/Validator/Tests/Fixtures/Entity.php]: 👍

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Pull request closed without merging**